test/e2e/set/scyllacluster/scyllacluster_tls.go

```diff
@@ -244,6 +249,128 @@ var _ = g.Describe("ScyllaCluster", func() {
+		if !utilfeature.DefaultMutableFeatureGate.Enabled(features.AutomaticTLSCertificates) {
+			g.Skip(fmt.Sprintf("Skipping because %q feature is disabled", features.AutomaticTLSCertificates))
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
+		defer cancel()
+
+		sc := scyllafixture.BasicScyllaCluster.ReadOrFail()
+		o.Expect(sc.Spec.Datacenter.Racks).To(o.HaveLen(1))
+		sc.Spec.Datacenter.Racks[0].Members = 1
+
+		framework.By("Creating a ScyllaCluster")
+		sc, err := f.ScyllaClient().ScyllaV1().ScyllaClusters(f.Namespace()).Create(ctx, sc, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Waiting for the ScyllaCluster to rollout (RV=%s)", sc.ResourceVersion)
+		waitCtx1, waitCtx1Cancel := utils.ContextForRollout(ctx, sc)
+		defer waitCtx1Cancel()
+		sc, err = utils.WaitForScyllaClusterState(waitCtx1, f.ScyllaClient().ScyllaV1(), sc.Namespace, sc.Name, utils.WaitForStateOptions{}, utils.IsScyllaClusterRolledOut)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		verifyScyllaCluster(ctx, f.KubeClient(), sc)
+		hosts := getScyllaHostsAndWaitForFullQuorum(ctx, f.KubeClient().CoreV1(), sc)
+		o.Expect(hosts).To(o.HaveLen(1))
+		host := hosts[0]
+
+		servingCABundleConfigMap, err := f.KubeClient().CoreV1().ConfigMaps(f.Namespace()).Get(ctx, fmt.Sprintf("%s-local-serving-ca", sc.Name), metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		servingCACerts, _ := verification.VerifyAndParseCABundle(servingCABundleConfigMap)
+		o.Expect(servingCACerts).To(o.HaveLen(1))
+
+		adminClientSecret, err := f.KubeClient().CoreV1().Secrets(f.Namespace()).Get(ctx, fmt.Sprintf("%s-local-user-admin", sc.Name), metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		_, adminClientCertBytes, _, adminClientKeyBytes := verification.VerifyAndParseTLSCert(adminClientSecret, verification.TLSCertOptions{
+			IsCA:     pointer.Ptr(false),
+			KeyUsage: pointer.Ptr(x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature),
+		})
+
+		servingCAPool := x509.NewCertPool()
+		servingCAPool.AddCert(servingCACerts[0])
+		adminTLSCert, err := tls.X509KeyPair(adminClientCertBytes, adminClientKeyBytes)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		const loaders = 3
+
+		for i := 0; i < loaders; i++ {
+			clusterConfig := gocql.NewCluster(hosts...)
+			clusterConfig.Timeout = 3 * time.Second
+			clusterConfig.ConnectTimeout = 3 * time.Second
+			clusterConfig.Port = 9142
+			clusterConfig.SslOpts = &gocql.SslOptions{
+				Config: &tls.Config{
+					ServerName:         host,
+					InsecureSkipVerify: false,
+					Certificates:       []tls.Certificate{adminTLSCert},
+					RootCAs:            servingCAPool,
+				},
+			}
+
+			di := insertAndVerifyCQLData(ctx, hosts, utils.WithClusterConfig(clusterConfig))
+			defer di.Close()
+
+			trafficCtx, trafficCtxCancel := context.WithCancel(ctx)
+			defer trafficCtxCancel()
+
+			stopTraffic, err := di.StartContinuousReads(trafficCtx)
```

Comment:
is there any reason not to use it in any other non disruptive test as well?

Reply:
(if we want to move it elsewhere I think it would be helpful to use an abstraction layer that reduces the injection point to a minimum)

Reply:
Test isolation, should test verifying upgrade fail when tls cert reload causes traffic disruption? It's easier to triage issues when test focuses on particular behavior, not on everything around.

Reply:
We have to make sure all of our operations are HA by default. That also means we'd have to double the number of tests. I think this should run by default in every test and could be checked at the end of the test so it can be clearly distinguished. But no non-disruptive test should be allowed to break HA invariant.

Reply:
there is also an option to use global invariants checkers and generate a synthetic ginko failure